src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -508,6 +522,49 @@ private bool ConfigureLocale()
             return true;
         }
 
+        private bool SyncOptionalWorkloads()
+        {
+            bool isHiveUpdated = false;
+            bool isCustomHive = _commandInput.HasDebuggingFlag("--debug:ephemeral-hive") || _commandInput.HasDebuggingFlag("--debug:custom-hive");
+
+            if (!isCustomHive)
+            {
+
+                string sdkVersion = EnvironmentSettings.Host.Version;
+
+                try
+                {
+                    List<InstallationRequest> oWInstallationRequest = new List<InstallationRequest>();
+                    HashSet<string> syncRequestsPackageIds = new HashSet<string>();
+                    TemplateLocator optionalWorkloadLocator = new TemplateLocator();
+
+                    IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> syncRequests = optionalWorkloadLocator.GetDotnetSdkTemplatePackages(sdkVersion);
+                    foreach (IOptionalSdkTemplatePackageInfo packageInfo in syncRequests)
+                    {
+                        oWInstallationRequest.Add(new InstallationRequest(packageInfo.Path, true));
+                        syncRequestsPackageIds.Add(packageInfo.TemplatePackageId);
+                    }
+
+                    if (oWInstallationRequest.Count != 0)
+                    {
+                        isHiveUpdated = true;
+                        Installer.InstallPackages(oWInstallationRequest, null, false, false);
+                    }
+
+                    /* TODO: add comparision of Optional SDK Workload templates from Hive and returned by TemplateLocator
+                     * implement templates removal if they are not returned by API anymore
```

Comment:
Just to double-check: When we address the TODO and add pulling existing templates from hive, will we be calling `Installer.InstallPackages()` only for packages not already in the cache or will we still be calling it for everything returned from `GetDotnetSdkTemplatePackages()` on each startup?

Reply:
We calling it for everything returned by API. The logic to check what needs to be installed and what needs to be updated, implemented inside the installer. I don't want to duplicate it in the `New3Command.cs`.
The performance would be the same since we just would move the checking from one place to another.

Reply:
Looks like you just duplicated it. Is there a reason we can't skip `owInstallationRequests.Add(...)` for packages that are already installed and have the same version?

Reply:
I put the `owInstallationRequests.Add(...)` under the condition when I am setting the `isHiveUpdated = true;`